tests/ui/type-alias-impl-trait/self-referential-3.rs

```diff
@@ -5,7 +5,7 @@
 type Bar<'a, 'b> = impl PartialEq<Bar<'a, 'b>> + std::fmt::Debug;
 
 fn bar<'a, 'b>(i: &'a i32) -> Bar<'a, 'b> {
```

Comment:
This is kind of a shame bc I think it just works in the new solver lol

Reply:
is that because of the new solver's handling of things that prove themselves?